src/overlays/actors/ovl_Item_Shield/z_item_shield.c

```diff
@@ -49,21 +40,192 @@ static ColliderCylinderInit D_80B871A0 = {
     },
     { 15, 15, 0, { 0, 0, 0 } },
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Item_Shield/func_80B86920.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Item_Shield/ItemShield_Init.s")
+const ActorInit Item_Shield_InitVars = {
+    ACTOR_ITEM_SHIELD,
+    ACTORCAT_ITEMACTION,
+    FLAGS,
+    OBJECT_LINK_CHILD,
+    sizeof(ItemShield),
+    (ActorFunc)ItemShield_Init,
+    (ActorFunc)ItemShield_Destroy,
+    (ActorFunc)ItemShield_Update,
+    (ActorFunc)ItemShield_Draw,
+};
+
+static Color_RGBA8 unused = { 255, 255, 0, 255 };
+static Color_RGBA8 unused2 = { 255, 0, 0, 255 };
+
+void func_80B86920(ItemShield* this, ItemShieldActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
```

Comment:
the standard name for this func is `<Actor>_SetupAction, if you wanted to name this one now